### PR TITLE
MEN-4486: [pipeline] Run tests in Hosted Mender staging cluster

### PR DESCRIPTION
**Pull request by lluiscampos**
Main commit:
* MEN-4486: [pipeline] Run tests in Hosted Mender staging cluster
This commit reworks the test job in the pipeline and the helper scripts
to connect to HM staging cluster and create a one time use namespace
where to install the chart and run the tests.

Other commits:
* [Makefile] Set shell to /bin/bash
As sh complains on invalid syntax
* [README] Document the obvious: build before install
* [ci-deps-k8s] Update URL for stable helm repository
The googleapis.com one is deprecated and prints a warning to switch to
the new one.
* [tests/] Split ci-deps-k8s.sh into two scripts: install and deploy
So that locally we can run the deploy one assuming all dependencies are
in place.
* [tests/ci-make-helm.sh] Use relative path instead of hardcoded CI path

### CODE REVIEW

**Comment by lluiscampos** [comment]
@tranchitella This PR brings my design to run the helm-charts in HM staging.

So far I haven't been successful. Creating the namespace is fine, next step is installing minio helm chart, and for that I added the following permissions into the hosted-mender-staging-ci `clusterrole`

```
  rule {
    api_groups = ["*"]
    resources  = ["serviceaccounts", "secrets", "roles", "rolebindings"]
    verbs      = ["create", "get", "list"]
  }
```

(I did `terraform apply` those while testing, but then reverted)

With that permissions it then [fails on update/patch secrets](https://gitlab.com/Northern.tech/Mender/mender-helm/-/jobs/1087074820) and servicemonitor, error follows:

```
Error: roles.rbac.authorization.k8s.io "mender-minio-update-prometheus-secret" is forbidden: user "hosted-mender-staging-ci" (groups=["hosted-mender-staging-ci" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
{APIGroups:[""], Resources:["secrets"], ResourceNames:["mender-minio-prometheus"], Verbs:["update" "patch"]}
{APIGroups:["monitoring.coreos.com"], Resources:["servicemonitors"], ResourceNames:["mender-minio"], Verbs:["get"]}
```

At this point I was a bit scared of compromising something and decided to ping you for help. My concern is that I don't know which on these resources (secrets, servicemonitors, possibly others) can escape a namespace, so that by update/patch might break or compromise something else in the cluster.

In short: please browse this PR and let me know if I am going in the right direction or not, and give me some advice on how to move it forward.

**Comment by lluiscampos** [comment]
Closing. I'll rework in a new PR
